lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -720,13 +863,15 @@ static void populateOpConversion(RewritePatternSet &patterns,
     ICmpOpConversion<WildcardNeOp, ICmpPredicate::wne>,
     
     // Patterns of structural operations.
-    SVModuleOpConversion, InstanceOpConversion,
+    SVModuleOpConversion, InstanceOpConversion, ProcedureOpConversion,
 
     // Patterns of shifting operations.
     ShrOpConversion, ShlOpConversion, AShrOpConversion,
 
     // Patterns of assignment operations.
-    AssignOpConversion<ContinuousAssignOp>,
+    AssignOpConversion<ContinuousAssignOp, 0, 1>,
+    AssignOpConversion<BlockingAssignOp, 0, 1>,
+    AssignOpConversion<NonBlockingAssignOp, 1, 0>,
```

Comment:
I noticed that `llhd.time` has `delta` and `epsilon`, I think they both are related to the precise, but the latter is more precise than the former, right? Therefore, `continuouse_assign` and `blokcing_assign` perform more quickly than `nonBlocking_assign` which will assign a value until the next delay :thinking:.

Reply:
Yes, that's correct. IEEE 1800-2017 §4.9 mentions that a non-blocking assign is scheduled for the NBA region while the blocking one is scheduled for the inactive region.